integration/build/build_test.go

```diff
@@ -696,6 +697,87 @@ func TestBuildPlatformInvalid(t *testing.T) {
 	assert.Assert(t, errdefs.IsInvalidParameter(err))
 }
 
+func TestBuildSecurityOpt(t *testing.T) {
+	ctx := context.Background()
+	defer setupTest(t)()
+
+	apiclient := testEnv.APIClient()
+
+	tests := []struct {
+		doc         string
+		skip        bool
+		securityOpt []string
+		expectedErr string
+	}{
+		{
+			doc: "no options",
+		},
+		{
+			doc:         "empty options",
+			securityOpt: []string{},
+		},
+		{
+			doc:         "linux: unsupported security-opt",
+			skip:        runtime.GOOS == "windows",
+			securityOpt: []string{"some-opt=some-value"},
+			expectedErr: "security options are not supported on " + runtime.GOOS,
+		},
+		{
+			doc:         "linux: credentialspec",
+			skip:        runtime.GOOS == "windows",
+			securityOpt: []string{"credentialspec=file://<anything>"},
+			expectedErr: "security options are not supported on " + runtime.GOOS,
+		},
+		{
+			doc:         "windows: invalid option",
+			skip:        runtime.GOOS != "windows",
+			securityOpt: []string{"option-without-value"},
+			expectedErr: "invalid security option: no equals sign in supplied value option-without-value",
+		},
+		{
+			doc:         "windows: credentialspec",
+			skip:        runtime.GOOS != "windows",
+			securityOpt: []string{"credentialspec=file://<anything>"},
+		},
+		{
+			doc:         "windows: credentialspec case-insensitive",
+			skip:        runtime.GOOS != "windows",
+			securityOpt: []string{"cReDeNtIaLsPeC=file://<anything>"},
+		},
+		{
+			doc:         "windows: unsupported security-opt",
+			skip:        runtime.GOOS != "windows",
+			securityOpt: []string{"credentialspec=file://<anything>", "seccomp=unconfined"},
+			expectedErr: "security option not supported: seccomp",
+		},
+	}
+	for _, tc := range tests {
+		if tc.skip {
+			continue
+		}
```

Comment:
I didn't use a `t.Skip()` to reduce the noise a bit, but happy to change if others prefer an explicit `t.Skip()`.